packages/consumption/src/modules/attributes/local/LocalAttribute.ts

```diff
@@ -151,14 +160,26 @@ export class LocalAttribute extends CoreSynchronizable implements ILocalAttribut
         return isPeerSharedAttribute;
     }
 
+    public isThirdPartyOwnedAttribute(ownAddress: CoreAddress, thirdPartyAddress?: CoreAddress): this is ThirdPartyOwnedRelationshipAttribute {
+        let isThirdPartyOwnedAttribute = this.isShared() && !this.isOwnedBy(ownAddress) && !this.isOwnedBy(this.shareInfo.peer);
+        if (!isThirdPartyOwnedAttribute) {
```

Comment:
This is implemented according to `isOwnSharedAttribute` and `isPeerSharedAttribute`. I think the idea is that we can return early, if the requirements aren't met in the first place. I doubt this saves much time/resources, however, it seems to be our coding style.

Reply:
It is not clear to me why this should return earlier, because in both implementation variants you would have to perform at least one check (in one case `if (!isThirdPartyOwnedAttribute)` and in the other case `if (typeof thirdPartyAddress !== "undefined")`). If `this.isShared() && !this.isOwnedBy(ownAddress) && !this.isOwnedBy(this.shareInfo.peer)` is `true`, two checks would currently have to be made, which is one unneccessary one. Only if `this.isShared() && !this.isOwnedBy(ownAddress) && !this.isOwnedBy(this.shareInfo.peer)` is `false` and `thirdPartyAddress` is defined, the variable `isThirdPartyOwnedAttribute` is currently prevented from being updated unnecessarily (from `false` to `false`). 

But you're right that this doesn't save much time/resources. And if it's our coding style, it doesn't need to be adapted here either.